File: cyanBat/src/main/java/at/msmiech/cyanbat/persistence/ScoreContract.java
package at.msmiech.cyanbat.persistence;

import android.provider.BaseColumns;

/**
 * Reference: Lecture
 *
 * @author kittysCode
 */
public class ScoreContract {

    /**
     * Provides constants for an entry in Score table. BaseColumns interface declares _ID = "id" constant.
     */
    public static abstract class ScoreEntry implements BaseColumns {
        public static final String TABLE_NAME = "highscore";

        public static final String COLUMN_TYPE_ID = "INTEGER PRIMARY KEY";

        public static final String COLUMN_NAME_USERNAME = "username";
        public static final String COLUMN_TYPE_USERNAME = "TEXT";

        public static final String COLUMN_NAME_SCORE = "score";
        public static final String COLUMN_TYPE_SCORE = "INTEGER";
    }

}

File: app/src/main/kotlin/at/smiech/cyanbat/persistence/ScoreContract.kt
package at.smiech.cyanbat.persistence

import android.provider.BaseColumns

/**
 * Reference: Lecture
 *
 * @author kittysCode
 */
class ScoreContract {

    /**
     * Provides constants for an entry in Score table. BaseColumns interface declares _ID = "id" constant.
     */
    abstract class ScoreEntry : BaseColumns {
        companion object {
            val TABLE_NAME = "highscore"

            val COLUMN_NAME_ID = "id"
            val COLUMN_TYPE_ID = "INTEGER PRIMARY KEY"

            val COLUMN_NAME_USERNAME = "username"
            val COLUMN_TYPE_USERNAME = "TEXT"

            val COLUMN_NAME_SCORE = "score"
            val COLUMN_TYPE_SCORE = "INTEGER"
        }
    }

}
